src/actions/getCourses.ts

```diff
@@ -0,0 +1,173 @@
+'use server'
+
+import type { Course } from '@/payload-types'
+import config from '@/payload.config'
+import * as Sentry from '@sentry/nextjs'
+import type { Where } from 'payload'
+import { getPayload } from 'payload'
+
+export interface GetCoursesParams {
+  offset?: number | null
+  limit?: number | null
+  types?: string | null
+  providers?: string | null
+  states?: string | null
+  affinityGroups?: string | null
+  modesOfTravel?: string | null
+  startDate?: string | null
+  endDate?: string | null
+}
+
+export interface GetCoursesResult {
+  courses: Course[]
+  hasMore: boolean
+  total: number
+  error?: string
+}
+
+export async function getCourses(params: GetCoursesParams): Promise<GetCoursesResult> {
+  const payload = await getPayload({ config })
+
+  try {
+    const { types, providers, states, affinityGroups, modesOfTravel, startDate, endDate } = params
+
+    const offset = params.offset || 0
+    const limit = params.limit || 10
+
+    const conditions: Where[] = []
+
+    if (types) {
+      const selectedTypes = types.split(',').filter(Boolean)
```

Comment:
Why does `getCourses()` want users to provide a comma-delimited string as opposed to an array here?